File: CPP/Sorting/Minimum Operations to Make a Uni-Value Grid.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        vector<int>nums;
        for(int i=0;i<grid.size();i++){
            for(int j=0;j<grid[0].size();j++){
                nums.push_back(grid[i][j]);
            }
        }
        sort(nums.begin(),nums.end());

        //check if making the uni-value is possible
        int base = nums[0];
        for(int i=0;i<nums.size();i++){
            if(abs(nums[i]-base)%x!=0)return -1;
        }

        // find median
        int mid = nums[nums.size()/2];

        int operation =0;
        for(auto num:nums){
            operation+=abs(num-mid)/x;
        }
        return operation;
    }
};

File: Java/Sorting/Minimum Operations to Make a Uni-Value Grid.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        ArrayList<Integer> nums = new ArrayList<Integer>();
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[0].length; j++) {
                nums.add(grid[i][j]);
            }
        }
        Collections.sort(nums);

        
        int base = nums.get(0);
        for (int i = 0; i < nums.size(); i++) {
            if (Math.abs(nums.get(i) - base) % x != 0)
                return -1;
        }
        int mid = nums.get(nums.size() / 2);
        int operation = 0;
        for (int num : nums) {
            operation += Math.abs(num - mid) / x;
        }
        return operation;
    }
}
